lib/routemaster/redis_broker.rb

```diff
@@ -25,18 +27,42 @@ def cleanup
       _cleanup
     end
 
+    # Allow to inject pre-built Redis clients
+    #
+    # Before storing a new connection, ensure that any previously
+    # set client is properly closed.
+    #
+    def inject(clients={})
```

Comment:
In principle I agree with you, and at the beginning I went for that design. Then I realized that the `Config` module just asks the `RedisBroker` singleton to intialize the connections.
The logic to initialize them (wrap them in the `Namespace` class, manage the lookup table, etc) is all in the `RedisBroker`, and the `Config` module is just a thin layer.